src/game/ui.py

```diff
@@ -102,17 +116,17 @@ def pieces_to_string(self, remaining_pieces, game_turn):
                 if piece_id >= 10:
                     display_string += ' '
                 display_string += '   '
-            display_string += '  '
+            display_string += ' '
         return display_string
 
     def selected_piece_to_string(self, piece_number, game_turn):
         if game_turn.selected_piece == piece_number:
             return "[" + str(piece_number) + "]"
         return " " + str(piece_number) + " "
 
-    def players_to_string(self, game_turn):
-        player_1 = self.selected_player_to_string('Player 1', game_turn.player_one_active)
-        player_2 = self.selected_player_to_string('Player 2', not game_turn.player_one_active)
+    def players_to_string(self, game_turn, players):
```

Comment:
In the case you've create a "Player" class, it would be possible to implement a "special" method (__str__) to achieve this task (cast an object to a string representation). This is the Java equivalent of toString().

More informations here: https://openclassrooms.com/courses/apprenez-a-programmer-en-python/les-methodes-speciales-1

Reply:
I thinked about it, but I prefer to keep all render methods in the UI class
This case : Changing UI does impact other games classes